subprojects/model-core/src/main/java/org/gradle/internal/reflect/annotations/impl/DefaultTypeAnnotationMetadataStore.java

```diff
@@ -349,6 +362,20 @@ private void processMethodAnnotations(Method method, Map<String, PropertyAnnotat
         }
     }
 
+    private void validateSetterForMutableType(PropertyAccessorType accessorType, Method method, ValidationErrorsBuilder errorsBuilder) {
+        Class<?> setterType = accessorType.propertyTypeFor(method);
+        if (isRedundant(setterType)) {
+            String propertyName = accessorType.propertyNameFor(method);
+            String typeName = setterType.getName();
+            errorsBuilder.recordError(String.format("'%s' of type '%s' has redundant setter method.", propertyName, typeName));
+        }
+    }
+
+    private boolean isRedundant(Class<?> setter) {
```

Comment:
The declaration of this method is confusing. What we want to answer here is: "should a setter for the given type be prohibited?"

```suggestion
    private boolean isSetterProhibitedForType(Class<?> propertyType) {
```

Reply:
Done